### PR TITLE
✨ Implement example embeds

### PR DESCRIPTION
**Pull request by matthiasrohmer**
Sorry for the delay, had multiple formatting/escaping/parsing issues that I didn't foresee. What's left: how do we want to deal with incomplete examples that only show a snippet but don't have a `<head>`? On amp.dev we use validator rules to add needed scripts but with bento that wouldn't work. Possible solutions I thought of:

1) Make all docs require at least one full example with a `<head>` that is automatically reused for all other examples on the page
2) Add frontmatter to the docs that describe needed resources and create the head automatically. Might possibly also make sense for a consolidated intro section as on amp.dev
3) Make docs authors require to define a head in a second, hidden code section

Bildschirmfoto 2021-10-11 um 18 34 58

### CODE REVIEW

File: site/_includes/layouts/example.njk [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<html>
+  <head>
+    <title>bento.dev Example: {{ title }}</title>
```
Comment (sebastianbenz) [inline site/_includes/layouts/example.njk:3]: nit, domain is going to be bentojs.dev

File: shortcodes/Example.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+const path = require('path');
+const Prism = require('prismjs');
+const md = require('markdown-it')();
+
+const examples = [];
+const heads = {};
+let counter = {};
+
+function exampleShortCode(nunjucksEngine) {
+  return new function () {
```
Comment (sebastianbenz) [inline shortcodes/Example.js:27]: Would a class work here as well?

**Review by sebastianbenz — APPROVED** [review]
Nice!

File: shortcodes/Example.js (hunk 2 above)
Comment (matthiasrohmer) [reply shortcodes/Example.js:26]: Sadly now. Don't quite know what nunjucks is doing under the hood.

**Pull request merged**